src/main.cpp

```diff
@@ -3092,8 +3090,8 @@ bool ActivateBestChain(CValidationState& state, const CBlock* pblock, bool fAlre
                 int nBlockEstimate = Checkpoints::GetTotalBlocksEstimate();
                 {
                     if (connman) {
-                        connman->ForEachNode([nBlockEstimate, hashNewTip](CNode* pnode) {
-                            if (chainActive.Height() > (pnode->nStartingHeight != -1 ? pnode->nStartingHeight - 2000 : nBlockEstimate)) {
+                        connman->ForEachNode([pindexNewTip, nBlockEstimate, hashNewTip](CNode* pnode) {
```

Comment:
nit:
would be good to standardize this and not input two different arguments from the same object. `pindexNewTip` arg, used to get the height, and `hashNewTip` are related to the same block.

We should pass only the primitive values or pass the block index pointer, not both. (I'm more inclined to pass both primitive values. We don't know how this will evolve in the future).
In other words, instead of passing `pindexNewTip` and removing `nNewHeight` from this function, input `nNewHeight`.